src/client.js

```diff
@@ -51,6 +52,101 @@ export class Saturn {
     this.loadNodesPromise = this._loadNodes(this.opts)
   }
 
+  /**
+   *
+   * @param {string} cidPath
+   * @param {object} [opts={}]
+   * @param {('car'|'raw')} [opts.format]
+   * @param {number} [opts.connectTimeout=5000]
+   * @param {number} [opts.downloadTimeout=0]
+   * @returns {Promise<object>}
+   */
+  async fetchCIDWithRace (cidPath, opts = {}) {
+    const [cid] = (cidPath ?? '').split('/')
+    CID.parse(cid)
+
+    const jwt = await getJWT(this.opts, this.storage)
+
+    const options = Object.assign({}, this.opts, { format: 'car', jwt }, opts)
+
+    if (!isBrowserContext) {
+      options.headers = {
+        ...(options.headers || {}),
+        Authorization: 'Bearer ' + options.jwt
+      }
+    }
+
+    const origins = options.origins
+    const controllers = []
+
+    const createFetchPromise = async (origin) => {
+      options.url = origin
+      const url = this.createRequestURL(cidPath, options)
+
+      const controller = new AbortController()
+      controllers.push(controller)
+      const connectTimeout = setTimeout(() => {
+        controller.abort()
+      }, options.connectTimeout)
+
+      try {
+        res = await fetch(parseUrl(url), { signal: controller.signal, ...options })
+        clearTimeout(connectTimeout)
+        return { res, url, controller }
+      } catch (err) {
+        throw new Error(
+          `Non OK response received: ${res.status} ${res.statusText}`
+        )
+      }
+    }
+
+    const abortRemainingFetches = async (res, controllers) => {
+      return controllers.forEach((controller) => {
+        if (res.controller !== controller) {
+          controller.abort('Request race unsuccessful')
+        }
+      })
+    }
+
+    const fetchPromises = Promise.any(origins.map((origin) => createFetchPromise(origin)))
+
+    const log = {
+      startTime: new Date()
+    }
+
+    let res, url, controller
+    try {
+      ({ res, url, controller } = await fetchPromises)
+
+      abortRemainingFetches(res, controllers)
+
+      const { headers } = res
+      log.url = url
+      log.ttfbMs = new Date() - log.startTime
```

Comment:
Record ttfbMs only if `res.ok`, as per https://github.com/filecoin-saturn/js-client/pull/24/files#diff-627fe08d8bba6a4fa76e7c5ed36ef6f16d126733e65e19236b391de0d34f1fe0R98

Reply:
I adjusted this, the other PR is no longer needed I think